datastream-server/src/main/java/com/linkedin/datastream/server/Coordinator.java

```diff
@@ -438,6 +438,7 @@ public void onDatastreamUpdate() {
             datastreamGroups.stream().filter(x -> x.getTaskPrefix().equals(task.getTaskPrefix())).findFirst();
         if (dg.isPresent()) {
           ((DatastreamTaskImpl) task).setDatastreams(dg.get().getDatastreams());
+          dg.get().getDatastreams().forEach(this::invokePostDataStreamStateChangeAction);
```

Comment:
yes, you are right. For the update, we do not have dedicated listener that runs only on a leader. I am thinking to make the postDS create, update, delete or state change not specific for DH publish. Like, if we want to allow any further followup actions, which requires task level knowledge, keeping logic here would be helpful and since we are running the post invocation logic as an async call, I thought it would be ok.